packages/frontend/src/common/components/Button.tsx

```diff
@@ -1,13 +1,40 @@
+import Link from 'next/link'
 import React from 'react'
 
 interface Props {
+  type?: 'primary' | 'secondary'
   title: string
+  loading?: boolean
+}
+interface LinkProps extends Props {
+  href: string
+  onClick?: never
+}
+
+interface ButtonProps extends Props {
+  onClick: () => void
+  href?: never
+}
+
+const classes = {
+  primary: 'btn-primary text-white',
+  secondary: 'bg-medium',
 }
 
-export function Button({ title }: Props) {
+export function Button({
```

Comment:
Nice. I like how we can reuse this component at various places even if we have to use `href` or `onClick` interchangeably.